server/src/test/Parsing.test.ts

```diff
@@ -313,6 +313,42 @@ implementation SentenceToGenerate
         });
     });
 
+    describe('switch rule', () => {
+        it('should parse without errors a switch with cases without parenthesis', (done) => {
+            checkInputValidityForRule(
+                (parser) => parser.instruction_switchCase_withValue(),
+                `
+switch( val ) {
+    case VALUE_1 : {
+        // do stuff
+    }
+    case VALUE_2 : {
+        // do something else
+    }
+}
+;`,
+            );
+            done();
+        });
+
+        it('should parse without errors a switch with cases between parenthesis', (done) => {
+            checkInputValidityForRule(
+                (parser) => parser.instruction_switchCase_withValue(),
+                `
+switch( val ) {
+    case (VALUE_1) : {
+        // do stuff
+    }
+    case (VALUE_2) : {
+        // do something else
+    }
+}
+;`,
+            );
+            done();
+        });
+    });
```

Comment:
If the styles can be mixed in a single _switch_, consider adding an example for that.

```suggestion
    });
    
        it('should parse without errors a switch with parentheses around some values', (done) => {
            checkInputValidityForRule(
                (parser) => parser.instruction_switchCase_withValue(),
                `
switch( val ) {
    case (VALUE_1) : {
        // do stuff
    }
    case VALUE_2 : {
        // do something else
    }
}
;`,
            );
            done();
        });
    });
```